src/plugins/vis_type_timeseries/server/lib/vis_data/helpers/check_aggs.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { AggNotSupportedInMode } from '../../../../common/errors';
+import { getAggsByType, AGG_TYPE } from '../../../../common/agg_utils';
+import { TSVB_METRIC_TYPES, TIME_RANGE_DATA_MODES } from '../../../../common/enums';
+import { Metric } from '../../../../common/types';
+
+export function isAggSupported(metrics: Metric[]) {
+  const parentPipelineAggs = getAggsByType<string>((agg) => agg.id)[AGG_TYPE.PARENT_PIPELINE];
+  const metricTypes = metrics.filter(
+    (metric) =>
+      parentPipelineAggs.includes(metric.type) || metric.type === TSVB_METRIC_TYPES.SERIES_AGG
+  );
+
+  if (metricTypes.length) {
+    throw new AggNotSupportedInMode(
+      metricTypes.map((metric) => metric.type).join(', '),
+      TIME_RANGE_DATA_MODES.ENTIRE_TIME_RANGE
```

Comment:
very nitpicky, and maybe for Stratoula to decide, but if it's easy maybe we could make it written with label and not key? I mean not 
`The aggregation derivative is not supported in entire_time_range mode`
but `The aggregation derivative is not supported in entire time range mode`

Reply:
Yes good one Marta. I prefer it like that

Reply:
but `AggNotSupportedInMode` is like a common mechanism which allow handle not supported aggs in any mode because if this I use value from `TIME_RANGE_DATA_MODES.ENTIRE_TIME_RANGE` as a part of text

Reply:
Instead of joining this with this why don't we use the label 'entire time range mode'?